pkg/config/config.go

```diff
@@ -0,0 +1,346 @@
+			ServiceDID:  presets.IndexingServiceDID.String(),
+			ServiceURL:  presets.IndexingServiceURL.String(),
+		},
+		Upload: UploadConfig{
+			ServiceDID: presets.UploadServiceDID.String(),
+			ServiceURL: presets.UploadServiceURL.String(),
+		},
+		Principals: make(map[string]string),
+	}
+}
+
+// fromCLI loads configuration values from CLI flags
+func fromCLI(ctx *cli.Context, cfg *Node) {
+	// Core settings
+	if ctx.IsSet("key-file") {
+		cfg.Core.KeyFilePath = ctx.String("key-file")
+	}
+	if ctx.IsSet("port") {
+		cfg.Core.ServerPort = ctx.Int("port")
+	}
+	if ctx.IsSet("public-url") {
+		cfg.Core.PublicURL = ctx.String("public-url")
+	}
+
+	// Directory settings
+	if ctx.IsSet("data-dir") {
+		cfg.Directories.DataDir = ctx.String("data-dir")
+	}
+	if ctx.IsSet("tmp-dir") {
+		cfg.Directories.TempDir = ctx.String("tmp-dir")
+	}
+
+	// Indexing settings
+	if ctx.IsSet("indexing-service-proof") {
+		cfg.Indexing.StorageProof = ctx.String("indexing-service-proof")
+	}
+	if ctx.IsSet("indexing-service-url") {
+		cfg.Indexing.ServiceURL = ctx.String("indexing-service-url")
+	}
+	if ctx.IsSet("indexing-service-did") {
+		cfg.Indexing.ServiceDID = ctx.String("indexing-service-did")
+	}
+	if ctx.IsSet("announce-url") {
+		cfg.Indexing.AnnounceURL = ctx.String("announce-url")
+	}
+
+	// Upload settings
+	if ctx.IsSet("upload-service-url") {
+		cfg.Upload.ServiceURL = ctx.String("upload-service-url")
+	}
+	if ctx.IsSet("upload-service-did") {
+		cfg.Upload.ServiceDID = ctx.String("upload-service-did")
+	}
+
+	// PDP settings
+	if ctx.IsSet("curio-url") {
+		if cfg.PDP == nil {
+			cfg.PDP = &PDPConfig{}
+		}
+		cfg.PDP.ServerURL = ctx.String("curio-url")
+	}
+	if ctx.IsSet("pdp-proofset") {
+		if cfg.PDP == nil {
+			cfg.PDP = &PDPConfig{}
+		}
+		cfg.PDP.ProofSet = ctx.Uint64("pdp-proofset")
+	}
+}
+
+// setupViperWithDefaults creates a new Viper instance with default values and environment bindings
+func setupViperWithDefaults() (*viper.Viper, error) {
+	v := viper.New()
+
+	// Set up environment variable binding
+	v.SetEnvPrefix("STORAGE")
+	v.AutomaticEnv()
+
+	// Define specific environment variable mappings
+	envMappings := map[string]string{
+		// Core
+		"core.key_file":   "PRIVATE_KEY",
+		"core.port":       "PORT",
+		"core.public_url": "PUBLIC_URL",
+
+		// Directories
+		"directories.data_dir": "DATA_DIR",
+		"directories.temp_dir": "TMP_DIR",
+
+		// Indexing
+		"indexing.storage_proof": "INDEXING_SERVICE_PROOF",
+		"indexing.service_did":   "INDEXING_SERVICE_DID",
+		"indexing.service_url":   "INDEXING_SERVICE_URL",
+		"indexing.announce_url":  "ANNOUNCE_URL",
+
+		// Upload
+		"upload.service_did": "UPLOAD_SERVICE_DID",
+		"upload.service_url": "UPLOAD_SERVICE_URL",
+
+		// PDP
+		"pdp.server_url": "CURIO_URL",
+		"pdp.proof_set":  "PDP_PROOFSET",
+	}
+
+	// Create the aliases for environment variables
+	for key, envVar := range envMappings {
+		if err := v.BindEnv(key, "STORAGE_"+envVar); err != nil {
```

Comment:
why do we need to use `BindEnv` if we have already used `AutomaticEnv`?

Reply:
ahh good catch, this binding call isn't necessary -- will remove.